allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -1100,7 +1100,8 @@ def stimulus_presentations(self) -> pd.DataFrame:
         table = table.rename(columns={"stop_time": "end_time"})
         # Backwards compatibility for data generated without the needed
         # ``stimulus_block`` column in the table.
-        if "stimulus_block" in table.columns:
+        if ("stimulus_block" in table.columns
```

Comment:
this logic to add the `trials_id` needs to be removed since it was added in the `Presentations` class

Reply:
We still need it unless we want to recreate the VBN data. Alternatively I can add it to the __init__ of the presentations class, though that would require that class to accept a `Trials` object.

Reply:
Okay, regarding this and the issue below. Likely the best way to do this (that is make Trials required and to accommodate data that doesn't have the column in the NWB) is to add a similar path `from_nwb` that passes the Trials object down the Stimuli and into Presentations to add the trials id. This one on `from_nwb` would be optional as we don't want to spend the time calculating if the `trials_id` already exists. Sound good?

Reply:
Slight change. Thinking about it, I think I just want `from_nwb` methods to just load what ever data is in the NWB, at lest for the data_object methods. Would you be happy with a method on `Presentations` that takes in a `Trials` data frame and adds the `trials_id` column if it doesn't already exist? This would make `from_nwb` a lot simpler at the cost of a public method on `Presentations` to add the column. The column would still be added as is in the `from_stimulus_file` method.

Reply:
My concern was that it was unclear why `trials_id` was being added in multiple places, and also it was unclear why `trials` was optional if we want to always calculate `trials_id`. I am ok with what you proposed though don't follow the difference in the separate public method approach vs just doing it in `from_nwb`.

Reply:
Okay, sorry, guess I over compensated. If I just add clarity via a comment and update to the docstring to the two places is that enough?